From 5344c8849a595535221da068f1fde3b0f1da8901 Mon Sep 17 00:00:00 2001
Message-Id: <5344c8849a595535221da068f1fde3b0f1da8901.1421835210.git.chang-joon.lee@intel.com>
In-Reply-To: <aa9849875fd9575e7de3d51e779ec326100b81c0.1421835210.git.chang-joon.lee@intel.com>
References: <aa9849875fd9575e7de3d51e779ec326100b81c0.1421835210.git.chang-joon.lee@intel.com>
From: Daniel Vetter <daniel.vetter@ffwll.ch>
Date: Wed, 26 Nov 2014 10:26:05 +0100
Subject: [PATCH 27/63] FROM_UPSTREAM [VPG]: drm/i915: Check locking in
 i915_gem_request_unreference

With refcounting it looks like you can just drop that refcount, but
that's not really the case. So make sure no one forgets.

Motivated by the unlocked call in the mmio flip code.

Change-Id: I4dda4da10d696a3ca78c051dce18a48b80e4de80
For: VIZ-4377
Cc: John Harrison <John.C.Harrison@Intel.com>
Cc: Thomas Daniel <Thomas.Daniel@intel.com>
Signed-off-by: Daniel Vetter <daniel.vetter@intel.com>
Upstream-repo: git://anongit.freedesktop.org/drm-intel
Upstream-commit: f245860ece9b18394b679c2c55738babc904f506
Upstream-as-of-tag: n/a
---
 drivers/gpu/drm/i915/i915_drv.h |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/i915/i915_drv.h b/drivers/gpu/drm/i915/i915_drv.h
index 6ae687a..8b7765c 100644
--- a/drivers/gpu/drm/i915/i915_drv.h
+++ b/drivers/gpu/drm/i915/i915_drv.h
@@ -2131,6 +2131,7 @@ i915_gem_request_reference(struct drm_i915_gem_request *req)
 static inline void
 i915_gem_request_unreference(struct drm_i915_gem_request *req)
 {
+	WARN_ON(!mutex_is_locked(&req->ring->dev->struct_mutex));
 	kref_put(&req->ref, i915_gem_request_free);
 }
 
-- 
1.7.9.5

